src/vm/inlinetracking.h

```diff
@@ -106,7 +106,7 @@ class InlineTrackingMapTraits : public NoRemoveSHashTraits <DefaultSHashTraits<I
     static count_t Hash(key_t k)
     {
         LIMITED_METHOD_DAC_CONTRACT;
-        return ((count_t)k.m_methodDef ^ (count_t)k.m_module);
+        return k.m_methodDef ^ *(count_t*)k.m_module;
```

Comment:
This converts k.m_module to a pointer and looks at the contents. The original code just wants to get the lowest 4 bytes of the pointer. I posted a fix for this here.

https://github.com/dotnet/coreclr/pull/22810/commits/5de5a8490e425685334cb4770fe3835492887232

Reply:
Thanks, I have it cherry-picked on my branch.